### PR TITLE
Migrate to checker framework

### PR DESCRIPTION
**Pull request by liach**
Signed-off-by: liach <liach@users.noreply.github.com>

See SpongePowered/SpongeAPI#2001

### CODE REVIEW

File: mcmod-info/src/main/java/org/spongepowered/plugin/meta/McModInfo.java [hunk 1]
```diff
@@ -24,14 +24,14 @@
  */
 package org.spongepowered.plugin.meta;
 
-import static java.util.Arrays.asList;
```
Comment (Minecrell) [inline mcmod-info/src/main/java/org/spongepowered/plugin/meta/McModInfo.java]: Our code style has static imports before regular imports: https://github.com/SpongePowered/SpongeAPI/blob/3fdbf2993d487e0170ce8f77823805e86fee254f/extra/intellij/sponge_intellij_style.xml#L38

**Comment by liach** [comment]
@Minecrell Fixed in next push.

File: mcmod-info/src/main/java/org/spongepowered/plugin/meta/McModInfo.java [hunk 2]
```diff
@@ -32,6 +32,8 @@
 import com.google.gson.JsonIOException;
 import com.google.gson.stream.JsonReader;
 import com.google.gson.stream.JsonWriter;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.framework.qual.DefaultQualifier;
```
Comment (Minecrell) [inline mcmod-info/src/main/java/org/spongepowered/plugin/meta/McModInfo.java]: This one looks unused

File: mcmod-info/src/main/java/org/spongepowered/plugin/meta/McModInfo.java [hunk 3]
```diff
@@ -54,6 +56,7 @@
  * Represents a serializer for {@link PluginMetadata} for the
  * {@code mcmod.info} file format.
  */
+@DefaultQualifier(NonNull.class)
```
Comment (Minecrell) [inline mcmod-info/src/main/java/org/spongepowered/plugin/meta/McModInfo.java]: Isn't this redundant given the one in `package-info.java` (although located in a different module)?

File: src/main/java/org/spongepowered/plugin/meta/version/package-info.java [hunk 4]
```diff
@@ -22,22 +22,5 @@
...
-        ElementType.FIELD,
-        ElementType.METHOD,
-        ElementType.PARAMETER
-})
-@Retention(RetentionPolicy.RUNTIME)
-public @interface NonnullByDefault {
-
-}
+@org.checkerframework.framework.qual.DefaultQualifier(org.checkerframework.checker.nullness.qual.NonNull.class)
+package org.spongepowered.plugin.meta.version;
```
Comment (Minecrell) [inline src/main/java/org/spongepowered/plugin/meta/version/package-info.java]: This specific package shouldn't be `@NonNull` by default. The reason is that it is directly sourced from the Maven source code with only little modifications. As such it is not appropriately annotated with `@Nullable` where necessary.

Comment (liach) [reply src/main/java/org/spongepowered/plugin/meta/version/package-info.java]: Done.

**Comment by liach** [comment]
@kashike Mind taking a look as well?

**Pull request merged**